test/functional/test_solution.py

```diff
@@ -0,0 +1,439 @@
+        current_state = method(*args)[key]
+        pp(current_state)
+        current_value = current_state[field]
+        while value != current_value:
+            sys.stdout.flush()
+            current_state = method(*args)[key]
+            pp(current_state)
+            current_value = current_state[field]
+            time.sleep(self.polling_interval)
+
+    def create_listener(self, lb_id):
+        listener_config =\
+        {'listener': {'name': 'test_listener',
+                      'loadbalancer_id': lb_id,
+                      'protocol': 'HTTP',
+                      'protocol_port': 80}}
+        return self.ncm.create_listener(listener_config)['listener']
+
+    def create_lbaas_pool(self, l_id):
+        pool_config = {'pool': {
+                       'name': 'test_pool_anur23rgg',
+                       'lb_algorithm': 'ROUND_ROBIN',
+                       'listener_id': l_id,
+                       'protocol': 'HTTP'}}
+        pool = self.ncm.create_lbaas_pool(pool_config)
+        return pool['pool']
+
+    def create_lbaas_member(self, p_id):
+        subnet_id = None
+        for sn in self.ncm.list_subnets()['subnets']:
+            if 'server-v4' in sn['name']:
+                subnet_id = sn['id']
+                break
+
+        member_config = {'member': {'subnet_id': subnet_id,
+                                    'address': self.symbols['server_client_ip'],
+                                    'protocol_port':
+                                        int(self.symbols['server_http_port'])}}
+
+        member = self.ncm.create_lbaas_member(p_id, member_config)
+        return member['member']
+
+    def create_lbaas_healthmonitor(self, p_id):
+        hm_config = {'healthmonitor': {'type': 'HTTP',
+                                       'delay': 10,
+                                       'timeout': 3,
+                                       'pool_id': p_id,
+                                       'max_retries': 2}}
+        # direct call to client, missing polling layer
+        hm = self.ncm.create_lbaas_healthmonitor(hm_config)
+        return hm['healthmonitor']
+
+    def create_proxy(self):
+        class Proxy(object):
+            def __init__(self):
+                self.loadbalancer = None
+                self.listener = None
+                self.lbaas_pool = None
+                self.members = []
+                self.healthmonitors = []
+
+        # create a fully-formed loadbalancer with VIP, members, etc. suitable
+        # for passing traffic between client and server
+        proxy = Proxy()
+        proxy.loadbalancer = self.create_loadbalancer()
+        proxy.listener = self.create_listener(proxy.loadbalancer['id'])
+        proxy.pool = self.create_lbaas_pool(proxy.listener['id'])
+        proxy.members.append(self.create_lbaas_member(proxy.pool['id']))
+        # lbaasv2 bug: lbaas_show_member does not return status attribute to
+        # know when member comes online.
```

Comment:
Did we file this bug for OpenStack or is it a known issue?  We should put the ID in here if it is.